distribution/src/main/resources/bin/elasticsearch-env

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+CDPATH=""
+
+SCRIPT="$0"
+
+# SCRIPT might be an arbitrarily deep series of symbolic links; loop until we
+# have the concrete path
+while [ -h "$SCRIPT" ] ; do
+  ls=`ls -ld "$SCRIPT"`
+  # Drop everything prior to ->
+  link=`expr "$ls" : '.*-> \(.*\)$'`
+  if expr "$link" : '/.*' > /dev/null; then
+    SCRIPT="$link"
+  else
+    SCRIPT=`dirname "$SCRIPT"`/"$link"
+  fi
+done
+
+# determine Elasticsearch home; to do this, we strip from the path until we find
+# bin, and then strip bin (there is an assumption here that there is no nested
+# directory under bin also named bin)
+ES_HOME=`dirname "$SCRIPT"`
+while [ "`basename "$ES_HOME"`" != "bin" ]; do
+  ES_HOME=`dirname "$ES_HOME"`
+done
+ES_HOME=`dirname "$ES_HOME"`
+
+# now make ES_HOME absolute
+ES_HOME=`cd "$ES_HOME"; pwd`
+
+# now set the classpath
+ES_CLASSPATH="$ES_HOME/lib/*"
+
+# now set the path to java
+if [ -x "$JAVA_HOME/bin/java" ]; then
+    JAVA="$JAVA_HOME/bin/java"
+else
+    JAVA=`which java`
+fi
+
+if [ ! -x "$JAVA" ]; then
+    echo "could not find java; set JAVA_HOME or ensure java is in PATH"
+    exit 1
+fi
+
+# check the JAVA version
+"$JAVA" -cp "$ES_CLASSPATH" org.elasticsearch.tools.JavaVersionChecker
+
+if [ $? -ne 0 ]; then
+    echo -n "the minimum required Java version is 8; "
+    echo "your Java version from $JAVA does not meet this requirement"
+    exit 1
+fi
+
+# don't let JAVA_TOOL_OPTIONS slip in (e.g. agents in Ubuntu); works around
+# https://bugs.launchpad.net/ubuntu/+source/jayatana/+bug/1441487
+if [ "x$JAVA_TOOL_OPTIONS" != "x" ]; then
+    echo "warning: ignoring JAVA_TOOL_OPTIONS=$JAVA_TOOL_OPTIONS"
+    unset JAVA_TOOL_OPTIONS
+fi
+
+# JAVA_OPTS is not a built-in JVM mechanism but some people think it is so we
+# warn them that we are not observing the value of $JAVA_OPTS
+if [ "x$JAVA_OPTS" != "x" ]; then
+    echo -n "warning: ignoring JAVA_OPTS=$JAVA_OPTS; "
+    echo "pass JVM parameters via ES_JAVA_OPTS"
+fi
+
+${source.path.env}
```

Comment:
@rjernst  No, this is correct. I define an expansion `source.path.env` in `distribution/build.gradle`; in the packaging distributions we want to source `/etc/default/elasticsearch` or `/etc/sysconfig/elasticsearch` but on in the archive distributions we want to to set `CONF_DIR` to `ES_HOME/config` if and only if `CONF_DIR` is not set so that's what `source.path.env` expands to there. Perhaps it could be named better, but that is the intent. Does that make sense to you?